torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1253,27 +1278,37 @@ def _get_key(self):
         """
         return (len(self.replacements), len(self.divisible))
 
-    def _produce_dyn_sizes(self, ex: torch.Tensor, source: Source) -> List[sympy.Expr]:
+    def _produce_dyn_sizes(self,
+                           ex: torch.Tensor,
+                           source: Source,
+                           dims: Dict[int, DIM_DYNAMISM_STATE]) -> List[sympy.Expr]:
+        if dims is None:
+            dims = {}
         from torch._dynamo.source import TensorPropertySource, TensorProperty
         size = []
         for i, val in enumerate(ex.size()):
-            is_dynamic = _is_dim_dynamic(ex, i)
-            if _should_allocate(is_dynamic, self.assume_static_by_default):
-                size.append(self.create_symbol(
-                    val, TensorPropertySource(source, TensorProperty.SIZE, i), is_dynamic
-                ))
+            if i not in dims:
+                dim_state = DIM_DYNAMISM_STATE.DUCK
             else:
-                size.append(sympy.Integer(val))
+                dim_state = dims[i]
+            size.append(self.create_symbol(
+                val, TensorPropertySource(source, TensorProperty.SIZE, i), dim_state
+            ))
         return size
 
-    def create_symbolic_sizes_strides_storage_offset(self, ex: torch.Tensor, source: Source):
+    def create_symbolic_sizes_strides_storage_offset(self,
+                                                     ex: torch.Tensor,
+                                                     source: Source,
+                                                     *,
+                                                     dynamic_dims: Dict[int, DIM_DYNAMISM_STATE],
+                                                     constraint_dims: Dict[int, "MinMaxConstraint"]):
```

Comment:
nit: constraint is not an adjective; `constrained_dims`

Reply:
https://github.com/pytorch/pytorch/pull/96012#discussion_r1126826538